src/Ledger.Net/LedgerManager.cs

```diff
@@ -119,6 +120,16 @@ public async Task<string> GetAddressAsync(uint account, uint index)
             return await GetAddressAsync(account, false, index, false);
         }
 
+        public async Task<EthereumAppSignTransactionResponse> SignTransactionAsync(uint account, uint index, bool isChange, byte[] transactionData)
+        {
+            byte[] derivationData = Helpers.GetDerivationPathData(CurrentCoin.App, CurrentCoin.CoinNumber, account, index, isChange, CurrentCoin.IsSegwit);
+
+            // Create base class like GetPublicKeyResponseBase and make the method more like GetAddressAsync
```

Comment:
There should probably be some shared base class for the sign transaction response, similar to the GetPublicKeyResponseBase. However, how the signed transaction data comes out and is handled is different between all the apps. So, the return type of the SignTransactionAsync method would be different every time. 

Maybe the method should have a generic return type of type Task<T> where T is some abstract base class SignTransactionResponseBase.